storage/connectors/jdbc/src/main/java/feast/storage/connectors/jdbc/retriever/SnowflakeQueryTemplater.java

```diff
@@ -146,22 +150,23 @@ protected String createJoinQuery(
     return writer.toString();
   }
 
-  // TODO: export as decoded csv file
   @Override
   protected List<String> generateExportTableSqlQuery(String resultTable, String stagingPath) {
-    String exportPath = String.format("%s/", stagingPath.replaceAll("/$", ""));
 
     List<String> exportTableSqlQueries = new ArrayList<>();
-    String createStageQuery = String.format("create or replace stage my_stage;");
+    // TODO: s3 integration needs to be configurable: s3_int
+    String createStageQuery =
+        String.format(
+            "create or replace stage my_s3_stage\n"
+                + "  storage_integration = s3_int\n"
+                + "  url = '%s';",
+            stagingPath);
     String copyIntoStageQuery =
         String.format(
-            "COPY INTO '@my_stage/%s.%s' FROM %s file_format = (type=csv compression='gzip')\n"
+            "COPY INTO '@my_s3_stage/%s.%s' FROM %s file_format = (type=csv compression='gzip')\n"
```

Comment:
it's probably better to do an adhoc unload without a stage, so it's not constantly replaced https://docs.snowflake.com/en/user-guide/data-unload-s3.html#unloading-data-directly-into-an-s3-bucket